<robot name="tf_simulation2">
    <material name="black">
        <color rgba="0 0 0 0.7"/>
    </material>
    <material name="white">
        <color rgba="1 1 1 0.7"/>
    </material>

    <link name="base_link">
        <visual>
            <geometry>
                <box size="0.3 0.2 0.1"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <material name="white" />
        </visual>
    </link>

    <joint name="base_to_ball_joint" type="fixed">
        <!-- <origin rpy="0 0 0" xyz="0 0.125 -0.05"/> -->
        <parent link="base_link"/>
        <child link="ball_link"/>
    </joint>
    <link name="ball_link">
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <geometry>
                <sphere radius="0.025"/>
            </geometry>
            <material name="white" />
        </visual>
    </link>

    <joint name="base_to_wheel_R" type="fixed">
        <!-- <origin rpy="0 -1.57079632679 0" xyz="0.125 -0.1 -0.025"/> -->
        <parent link="base_link"/>
        <child link="wheel_R"/>
    </joint>
    <link name="wheel_R">
        <visual>
            <geometry>
                <cylinder length="0.05" radius="0.05"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <material name="black"/>
        </visual>
    </link>

    <joint name="base_to_wheel_L" type="fixed">
        <!-- <origin rpy="0 -1.57079632679 0" xyz="-0.125 -0.1 -0.025"/> -->
        <parent link="base_link"/>
        <child link="wheel_L"/>
    </joint>
    <link name="wheel_L">
        <visual>
            <geometry>
                <cylinder length="0.05" radius="0.05"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <material name="black"/>
        </visual>
    </link>

    <joint name="laser_joint" type="fixed">
        <parent link="base_link"/>
        <child link="laser_link"/>
        <!-- <origin rpy="0 0 1.57079632679" xyz="0 0.13 0.07"/> -->
    </joint>
    <link name="laser_link">
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <geometry>
                <cylinder length="0.04" radius="0.02"/>
            </geometry>
            <material name="rad">
                <color rgba="1 0 0 0.7"/>
            </material>
        </visual>
    </link>
</robot>

